src/meta/src/hummock/hummock_manager.rs

```diff
@@ -1140,14 +1145,25 @@ where
     }
 
     pub async fn proceed_version_checkpoint(&self) -> risingwave_common::error::Result<()> {
-        let versioning_guard = self.versioning.read().await;
+        let mut versioning_guard = self.versioning.write().await;
```

Comment:
It is not good to hold the versioning write lock while doing I/O on meta store. Although we did the same thing in other methods, I suggest here we only acquire the write lock when we need to delete the in-memory `HummockVersionDelta` after we finish deleting `HummockVersionDelta` on meta store.

Reply:
> It is not good to hold the versioning write lock while doing I/O on meta store. Although we did the same thing in other methods, I suggest here we only acquire the write lock when we need to delete the in-memory `HummockVersionDelta` after we finish deleting `HummockVersionDelta` on meta store.

but if we put them in a VarTransaction, how can we split them?

Reply:
`VarTransaction` is not suitable for this use case. We can split first and then delete all entries in meta store via a meta store transaction (`delete_in_transaction` and then `commit_trx`)

Reply:
> `VarTransaction` is not suitable for this use case. We can split first and then delete all entries in meta store via a meta store transaction (`delete_in_transaction` and then `commit_trx`)

fixed